stock_measuring_device_zippcube/components/zippcube_device_component.py

```diff
@@ -0,0 +1,54 @@
+# Copyright 2021 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
+
+import logging
+
+from odoo import _
+
+from odoo.addons.component.core import Component
+
+_logger = logging.getLogger(__name__)
+
+
+class ZippcubeDevice(Component):
+    _name = "zippcube.device.component"
+    _inherit = "measuring.device.component"
+    _usage = "zippcube"
+
+    def _update_packaging_measures(self, measures):
+        self.collection.ensure_one()
+        packaging = self.env["product.packaging"].search(
+            [("measuring_device_id", "=", self.collection.id)], order="write_date DESC"
+        )
+        self._validate_packaging(packaging)
+        packaging_to_update = packaging[0]
+
+        self.env["measuring.wizard.line"].flush(
+            ["packaging_id", "wizard_id", "scan_requested"]
+        )
+        self.env["measuring.wizard"].flush(["device_id"])
+        wizard_line = self.collection.env["measuring.wizard.line"].search(
```

Comment:
the whole part about finding the line (line 26-38) should stay on the wizard IMO or on the device.
Something like `self.collection._get_line_to_scan()`.
Note: on the abstract component having a property `device` that return `self.collection` could make things more clear.